lib/AST/DiagnosticEngine.cpp

```diff
@@ -93,19 +93,13 @@ static_assert(sizeof(storedDiagnosticInfos) / sizeof(StoredDiagnosticInfo) ==
               "array size mismatch");
 
 static constexpr const char * const diagnosticStrings[] = {
-#define ERROR(ID, Options, Text, Signature) Text,
-#define WARNING(ID, Options, Text, Signature) Text,
-#define NOTE(ID, Options, Text, Signature) Text,
-#define REMARK(ID, Options, Text, Signature) Text,
+#define DIAG(KIND, ID, Options, Text, Signature) Text,
```

Comment:
Based on the description of the issue, seems like there are other places where this can also be done :) 
Or is this the only place @HassanElDesouky?

Reply:
@LucianoPAlmeida Good point! I see that the only other place that it can be added is here: https://github.com/apple/swift/blob/7db8b0f53760e527c7a42f1d8e7c394b2198efe4/lib/AST/DiagnosticEngine.cpp#L80

Reply:
I think if you do a text seach for something like `#define WARNING` using a text editor you find some other places :)

Reply:
Noted, I actually did search it, in Xcode, but missed that one, thanks for pointing out :)

Reply:
@LucianoPAlmeida I think, we shouldn't make any changes if only a subset of the diagnostic types (error/warning/note/remark) are defined. So, after @so5462 modify `storedDiagnosticInfos[]` I think it will be it. :)

Reply:
@HassanElDesouky maybe I am misunderstanding something here, since this one is specifically for categorizing based on DiagnosticKind here, since this is expecting an enum, does that mean KIND will have to be checked as well? I thought this would be a case, where we wouldn't have to DIAG